python/pyspark/sql/connect/client.py

```diff
@@ -531,12 +560,16 @@ def _analyze(self, plan: pb2.Plan, explain_mode: str = "extended") -> AnalyzeRes
             req.explain.explain_mode = pb2.Explain.ExplainMode.CODEGEN
         else:  # formatted
             req.explain.explain_mode = pb2.Explain.ExplainMode.FORMATTED
-
         try:
-            resp = self._stub.AnalyzePlan(req, metadata=self._builder.metadata())
-            if resp.client_id != self._session_id:
-                raise SparkConnectException("Received incorrect session identifier for request.")
-            return AnalyzeResult.fromProto(resp)
+            for attempt in Retrying(SparkConnectClient.retry_exception, **self._retry_policy):
+                with attempt:
+                    resp = self._stub.AnalyzePlan(req, metadata=self._builder.metadata())
+                    if resp.client_id != self._session_id:
+                        raise SparkConnectException(
+                            "Received incorrect session identifier for request."
```

Comment:
Maybe providing `resp.client_id` and `self._session_id` into error messages would helpful? Or maybe is it just verbose?

Reply:
This is unchanged code from the old handling, but I can add the two IDs

Reply:
Done.